src/csharp/Microsoft.Spark/Sql/Types/ComplexTypes.cs

```diff
@@ -71,14 +71,72 @@ private DataType FromJson(JObject json)
     }
 
     /// <summary>
-    /// The data type for a map. This class is not implemented yet.
+    /// The data type for a map. 
     /// </summary>
     public sealed class MapType : DataType
     {
         /// <summary>
-        /// Initializes the <see cref="MapType"/> instance.
+        /// Constructor for MapType class.
         /// </summary>
-        public MapType() => throw new NotImplementedException();
+        /// <param name="keyType">The data type of keys in this map</param>
+        /// <param name="valueType">The data type of values in this map</param>
+        /// <param name="valueContainsNull">Indicates if values can be null</param>
+        public MapType(DataType keyType, DataType valueType, bool valueContainsNull = true)
```

Comment:
any reason this defaults to `true`?

Reply:
Spark sets up this defaults to `true`. Please refer to https://github.com/apache/spark/blob/2b24a71fec021755f43db99628a56bd4a01518eb/sql/catalyst/src/main/scala/org/apache/spark/sql/types/MapType.scala#L97

Reply:
Got it. thanks.